### DISCUSSION TITLE
Escape key not working when bound to caps lock (Linux)

### DISCUSSION THREAD

**Discussion by ronandalton** [ppy/osu#14839] [Q&A] [answered]
I bind my escape key to caps lock so it's easier to reach.
The command to do this on Ubuntu is `setxkbmap -option "caps:escape"`
Pressing the caps lock key while in game isn't registered as pressing escape.
All other programs and games I've tested don't have this issue.

**Comment by peppy** [comment, marked as answer]
Have you tried changing your key binding in the game?

**Reply by ronandalton** [reply]
Thanks for the quick reply. Yes that does indeed work, thanks!
Other programs don't require manually setting this keybind though. Is this because osu hooks into the keyboard at a lower level?
Also, does this mean that the keyboard layout is ignored altogether? If so then I can see that being an issue beyond just wanting to swap the caps and escape key.

Thanks.

**Reply by peppy** [reply]
I believe this is due to the way we are reading key codes. I'd need to double check how things work on windows/macOS to make any calls here (as to whether it's something our end or linux specific), but you can find some more recent discussion at https://github.com/ppy/osu-framework/issues/4618.

Safe to say we're aware things are not perfect in all cases, but you can work around the issues for the time being.
